app/src/main/java/org/oppia/android/app/profile/AddProfileActivity.kt

```diff
@@ -44,4 +44,14 @@ class AddProfileActivity : InjectableAppCompatActivity() {
     super.onDestroy()
     addProfileFragmentPresenter.dismissAlertDialog()
   }
+
+  override fun onSaveInstanceState(outState: Bundle) {
+    addProfileFragmentPresenter.handleOnSavedInstanceState(outState)
+    super.onSaveInstanceState(outState)
+  }
+
+  override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
+    super.onRestoreInstanceState(savedInstanceState)
+    addProfileFragmentPresenter.handleOnRestoreInstanceState(savedInstanceState!!)
+  }
```

Comment:
Using `onSaveInstanceState` we can survive configuration changes, but what if the app keeps running for a long time in the background(maybe for a day), and then we open the app from "recent apps", will this still work?

Reply:
@prayutsu I don't understand what your doubt is
In most cases when the app is in the background for a day the app will restart
else there will be no issues but it has nothing to do with onSavedInstance

Correct me if I'm wrong

Reply:
Actually, I was more specifically talking about `System-initiated process death` and I think `onSavedInstanceState()` survives it easily.